test/testcases/test_http_api/test_session_management/test_session_sdk_routes_unit.py

```
@@ -902,6 +903,23 @@ def _raise_lookup(*_args, **_kwargs):
+@pytest.mark.p2
+@pytest.mark.parametrize(
+    ("release_value", "expected_release_mode"),
+    [(True, True), (False, False), ("true", True), ("TRUE", True), ("false", False), ("", False)],
+)
+def test_create_agent_session_parses_release_flag_without_string_coercion(release_value, expected_release_mode):
+    repo_root = Path(__file__).resolve().parents[4]
+    source = (repo_root / "api" / "apps" / "restful_apis" / "agent_api.py").read_text(encoding="utf-8")
+    tree = ast.parse(source)
+    parser_node = next(node for node in tree.body if isinstance(node, ast.FunctionDef) and node.name == "_parse_release_flag")
+    namespace = {}
+    exec(compile(ast.Module(body=[parser_node], type_ignores=[]), "agent_api.py", "exec"), namespace)
```

Ruff (0.16.2)

[error] Use of `exec` detected

(S102)